sycl/source/CMakeLists.txt

```diff
@@ -4,9 +4,17 @@
 #cmake_policy(SET CMP0057 NEW)
 #include(AddLLVM)
 
+include_directories(${LLVM_EXTERNAL_XPTI_SOURCE_DIR}/include)
+
 function(add_sycl_rt_library LIB_NAME)
+  # Add an optional argument so we can get the library name to
+  # link with for Windows Debug version
+  cmake_parse_arguments(ARG "" "XPTI_LIB" "" ${ARGN})
+  if (NOT DEFINED ARG_XPTI_LIB)
+    set(ARG_XPTI_LIB xpti)
+  endif()
```

Comment:
This code is primarily to handle Windows builds - need to build a static library with a different name to avoid build failures.  At the link stage, it links in the right library.

Reply:
@tovinkere this doesn't sound good. Can you please describe the error in more detail?

Reply:
@alexbatashev  When this patch was not present, the CI would sometimes link the SYCL runtime with a wrong version of xpti library. - as in Debug library with Release and vice-versa. This patch force fixed by explicitly sending the library name.

Reply:
@alexbatashev  This has been refactored now, so please take a look.